src/__tests__/shared/components/input/TextFieldWithLabelFormGroup.test.tsx

```diff
@@ -18,101 +19,89 @@ describe('text field with label form group', () => {
         />,
       )
 
-      const label = wrapper.find(Label)
-      expect(label).toHaveLength(1)
-      expect(label.prop('htmlFor')).toEqual(`${expectedName}TextField`)
-      expect(label.prop('text')).toEqual(expectedName)
+      expect(screen.getByText(expectedName)).toHaveAttribute('for', `${expectedName}TextField`)
     })
 
     it('should render label as required if isRequired is true', () => {
-      const expectedName = 'test'
-      const expectedRequired = true
-      const wrapper = shallow(
+      render(
         <TextFieldWithLabelFormGroup
-          name={expectedName}
+          name="test"
           label="test"
           value=""
           isEditable
-          isRequired={expectedRequired}
+          isRequired
           onChange={jest.fn()}
         />,
       )
 
-      const label = wrapper.find(Label)
-      expect(label).toHaveLength(1)
-      expect(label.prop('isRequired')).toBeTruthy()
+      expect(screen.getByTitle(/required/i)).toBeInTheDocument()
```

Comment:
Not sure if this was the best way to test if the field was required, but without the `prop` I struggled to find alternatives.

[Here is the playground](https://testing-playground.com/#markup=DwEwlgbgfKkAQGMA2BDAzmgvAIgGYHsAnAWwFoBzQ-AVwAdsZxphUAjAUyUVQxwJNJtO2OPxwAXdmnEAVdgA9xAMTCcQI8WHFJ2OGQAswaOEbgo4hdgEdqYSyBMA7WtXENJ04GDjSAnjpwEfCQiAC4LdhAAbgZgNAhyM0IwFFJDEBB2RwlCanYRAgRqNBRWALwUJDR8uBAUcVTaS1wweT50ETqG0jAg7Ox0SWS0AGsRZHQsbHjyHsckMEd2UlJcczXSQfZhkdFUgHdSAEYANjgRKnKwYnIReWIkRyn9cXFaUIB6D-2fgDp9gDMvyI5A+ACYAAxQj4zEQQVT7ABC+Da2AhcHRAFYjmC4NiwSI-OV9OwwOQXuEAOy0eRROAQbaaBCVTYLciOcLifC0OnEFCEciLQTsXDicJgmkxGC0er6URgJBIQLUQiWRziADCwSInRwAFkACyUgAcvzBRzgAIBBt+EIAnACADJWs5gzEnJBHA1gs1HUgm20OhBHI6-Sl2zGkE6-I2UuBHTG-O0nP3m34A40GuB2s0AsGkK22gGUpDHHMG40FoGUg1HBBRsNgysh34nTEA0hpyn5hMxu12qtmsGU0imzFgp1g5Px43RylHY1IIFGyOnH2Qg0aqc5yknAHx0MQlNwJumh1x9HDxMVjH6Atnk71o7L90Yzs2pv7ltH1NA-tZsEbWxA1HXNM4FznBc4EpSkzTbeMIR9E4IUpJ9Q0xI8GztXdO2jA1+wLX5n0HMFhzgMcJ1LI5y2NS1q1res5ybA9W3bTtfndXEwWjV8cwnfdCwhYtHSOXcT1fQtwzom17QBNCwwjOA8Jg45E2TC00wzA1SD4vM6KLEtqJjWjJIYpjaJbNt9y7c142vft9NIuMKMnU5LTdM1jTBUtl0pTF4xOdcIU3c112LPEoSTAE7RPCdbROLN8RPAEeNxfF9AzJMU2DF8zkvD9jQ7b9jzi-tI0AjivVA6c80TJtvOfU1KQBfzMyIkNg3Qo8lKTMTuL7GKgWfRzh1HDjKKMitBxrOtzOOUMrJPDi91wjjMQHXT80ElKAC9Yg+GVxH0GAYQSE6wBOoQkBOphGHgCZeDwIgyEoGh6BgSRFH5dhzCofYpgNcYeCmfgyD6cRLn2z6GksFA7ogbhJhwRYIEqMAQFWdhIlYFAEDGG7IAJ5gPlu4AScJoA) if anyone want's to make a suggestion?

Reply:
This feels more sensible.
`data-icon="asterisk"` more implementation details.

Reply:
Just clarifying, do you mean what I have is the more sensible option?

Reply:
yes I think so